dynamicio/core.py

```diff
@@ -227,6 +227,16 @@ def _apply_schema(self, df: pd.DataFrame) -> pd.DataFrame:
             raise ColumnsDataTypeError()
         return df
 
+    @staticmethod
+    def _transform_class_names_to_dataset_names(string_to_transform: str) -> str:
+        """Called by the __innit__ function to fetch dataset names from class name.
+
+        Used to create dataset names from class names, turns camel case into upper snake case.
+        For example: 'ThisNameABC' -> 'THIS_NAME_ABC'.
+        """
+        words = re.findall(r"\d[A-Z]+|[A-Z]?[a-z\d]+|[A-Z]{2,}(?=[A-Z][a-z]|\d|\W|$)|\d+|[A-Z]{2,}|[A-Z]", string_to_transform)
```

Comment:
There @tylerferguson now this beauty up there makes this `"READ_S3_PARQUET_I_O"` into this `"READ_S3_PARQUET_IO"`